task(settings,graphql): Create example of MFA guard using gql queries

dschom opened the pull request:
## Because

- We also wanted to have an example of using MFA with gql queries

## This pull request

- Adds a new test page `/settings/mfa_guard/test/gql`
- Adds a new mutation to simulate a JWT protected call being proxied to auth-server
- Shows how to deal with GQL error so MFA modal pops up


## Issue that this pull request solves

Closes: FXA-12324

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Same as before, to test:
- Create account
- Once on settings page navigate to `/settings/mfa_guard/test/gql`

Behavior should be the same as `/settings/mfa_guard/test/auth-client` page. Example over [here](https://github.com/mozilla/fxa/pull/19383#issuecomment-3247235595).

LZoog commented on packages/fxa-settings/src/components/Settings/PageMfaGuardWithGqlTest/index.tsx:
```diff
@@ -0,0 +1,198 @@
+  }, [jwtCache, mfaTest, errorHandler, sessionToken]);
+
+  // Wrap the page's content with an MfaGuard to protect it from access without
+  // a JWT that has a scope of "test"
```
This comment should probably go where we use `MfaGuard`?

chenba reviewed (approved):
I can't switch branches to try it out at the moment, but looks fine

LZoog commented on packages/fxa-settings/src/components/Settings/PageMfaGuardWithGqlTest/index.tsx:
```diff
@@ -0,0 +1,198 @@
+
+const TestWithGql = (_: RouteComponentProps) => {
+  const errorHandler = useErrorHandler();
+  const jwtCache = useSyncExternalStore(
```
Nice 👍

LZoog commented on packages/fxa-settings/src/components/Settings/PageMfaGuardWithGqlTest/index.tsx:
```diff
@@ -0,0 +1,198 @@
+
+export const PageMfaGuardTestWithGql = (props: RouteComponentProps) => {
+  return (
+    <MfaGuard requiredScope="test">
```
<img width="1596" height="833" alt="image" src="https://github.com/user-attachments/assets/ca7dc161-063b-4f8f-905d-736991cd5e9b" />

I can't seem to get around this whether I input a valid code or an invalid code 🤔 But it seems to work okay for the auth-client version.

LZoog reviewed (commented):
Code-wise it looks fine/good to me! I posted a screenshot though where it seems to just error out?

Here are a few general comments too that might've been better on your other PR:
* We don't design for super old iPhone models but I think we may want to at least make the modal scrollable. Also, ignore this if you're referencing Figma designs but should we make the modal just a smidge wider at tablet+? I think it might look nice. I love the big CTA buttons though.  <img width="549" height="670" alt="image" src="https://github.com/user-attachments/assets/e659cca1-9079-483d-9e4a-04d8a641c9b4" />
* Might be an existing thing: do we want to disallow non-numbers from being inputted and keep it at 6 digits?
* Looks to be an existing issue: we need `resend-code-success-heading` added to an FTL file
* Can we get the code showing up when running pm2 logs inbox instead of the full email HTML?

LZoog reviewed (approved):
LGTM! All my comments are fine as follow ups.

dschom commented:
@LZoog Thanks for the feedback.

- Filed follow up for Making modal scrollable FXA-12334
- Filed follow up for restricting input box to numeric values FXA-12335
- Fixed pm2 log output (good recommendation)
- Fixed issue with l10n-id in banner (The string was already in the .ftl file)
